src/main/java/de/tum/cit/aet/artemis/programming/service/RepositoryService.java

```diff
@@ -153,12 +154,17 @@ public Map<String, String> getFilesContentAtCommit(ProgrammingExercise programmi
      * This method filters out all non-file type entries and reads the content of each file.
      * Note: If an I/O error occurs reading any file, this exception is caught internally and logged.
      *
-     * @param repository The repository from which files are to be fetched.
+     * @param repository   The repository from which files are to be fetched.
+     * @param omitBinaries omit binary files for brevity and reducing size
      * @return A {@link Map} where each key is a file path (as a {@link String}) and each value is the content of the file (also as a {@link String}).
      *         The map includes only those files that could successfully have their contents read; files that cause an IOException are logged but not included.
      */
-    public Map<String, String> getFilesContentFromWorkingCopy(Repository repository) {
-        var files = gitService.listFilesAndFolders(repository).entrySet().stream().filter(entry -> entry.getValue() == FileType.FILE).map(Map.Entry::getKey).toList();
+    public Map<String, String> getFilesContentFromWorkingCopy(Repository repository, boolean omitBinaries) {
+        var files = gitService.listFilesAndFolders(repository, omitBinaries).entrySet().stream().filter(entry -> entry.getValue() == FileType.FILE).map(Map.Entry::getKey).toList();
+        return getFileListWithContent(files);
+    }
+
+    private Map<String, String> getFileListWithContent(List<File> files) {
```

Comment:
This method seems to be unnecessary (it can stay included in getFilesContentFromWorkingCopy).